### ISSUE TITLE
Checkpoints for baseline networks?

### ISSUE THREAD

**Issue by VSehwag** [google-research/simclr#10]
Could it possible to also release the checkpoints for the supervised networks (especially ResNet-4x), which are used as a baseline throughout the paper? Except for ResNet50, the checkpoints for larger width nets are unavailable at tensorhub or any other pre-trained network sources.

**Comment by chentingpc** [comment]
Yes it is possible, but there is no guarantee on the timeline. I will
update here when it became available.

Best,
Ting

**Comment by VSehwag** [comment]
Just a gentle reminder. Appreciate it if you make these models available to the community.

**Comment by chentingpc** [comment]
I will close this for now since there has not been enough efforts to release those supervised checkpoints  and managed them in this repo. I will comment back if there's any update. It is also not difficult to repurpose the current code for supervised learning (simply using the fine-tuning mode), so you can train one if you need to.

**Comment by jlindsey15** [comment]
Hello!  I wanted to ask if by any chance there is any update on this.  It would be very helpful for me to be able to compare SimCLR-trained models and supervised-trained models of equal size.  Training a ResNet50-4x on an 8-GPU setup is quite slow and requires a pretty small batch size (I'm getting less than 2 epochs per day).  I imagine that cloud TPUs would solve this problem, but I'd rather not go that route if possible due to limited resources.  If the model you used is shareable it would be greatly appreciated!  But I totally understand if that's not possible.  Just wanted to check.

Thanks!

**Comment by chentingpc** [comment]
Hi, we're in the process of opensourcing simclr v2 checkpoints (Table 1 in https://arxiv.org/pdf/2006.10029.pdf), which we plan to include supervised models. hopefully the supervised models there could be helpful enough?

**Comment by jlindsey15** [comment]
That's good to know, thanks!  That will be very helpful.  It would still also be useful to have the 4x wide version available, but I totally understand if there are other priorities, and I really appreciate that you are releasing the SimCLRv2 checkpoints!

**Comment by chentingpc** [comment]
The SimCLRv2 checkpoints just became available (including supervised model of different architectures)! Let me know if you have further questions.

**Comment by jlindsey15** [comment]
Awesome!  One question: some of the models have the following signatures:

['default', 'block_group1', 'block_group2', 'block_group3', 'block_group4', 'final_avg_pool', 'initial_max_pool', 'logits_sup', 'initial_conv']

While others have these:

['default', 'logits_sup', 'proj_head_input', 'proj_head_output']

For the second kind, what is the easiest way to extract intermediate layer outputs of the model?

**Comment by chentingpc** [comment]
Supervised networks are trained with standard argumentation (i.e. no color or blur as in SimCLR augmentation).

**Comment by jlindsey15** [comment]
Ok thanks!  Could you confirm the batch size that was used? Or the command used to train one of them if it's not a hassle?

**Comment by chentingpc** [comment]
Yeah, they are trained with batch size of 4096, except for ResNet-152 (3x+SK), which uses a smaller batch size. The batch size doesn't matter much for supervised learning (especially when they're <= 4096).

**Comment by jlindsey15** [comment]
Hi!  Another question along these lines.  I am doing some analyses that involve comparing the representations learned by self-supervised networks (like SimCLR and MOCO) vs supervised ones.  But I notice there are systematic differences in various aspects of the representations of networks in this repo (SimCLR, SimCLR v2, and the provided supervised baselines) and those from PyTorch code (MOCO pretrained models, and pretrained supervised models from Torchvision).  For instance, the activations of models produced by this repo are consistently much sparser (even comparing the supervised network from this repo to the supervised network from Torchvision).  Do you have any idea how the training code in this repo might deviate from PyTorch defaults to produce such differences?

**Comment by chentingpc** [comment]
Thanks for bringing up the observation. Just to clarify, how much sparsity is due to the implementation, rather than model training (i.e. self-supervised vs supervised)? There are clearly many differences between self-supervised and supervised. But in terms of implementation difference of supervised learning between this repo and pytorch vision, there seems to be  small things (according my limited familiarity with torchvision) - the learning rate schedule and normalization in input preprocessing.

**Comment by jlindsey15** [comment]
I think it can be mostly/entirely attributed to the implementation -- the gap between the two supervised networks is similar to the gap between SimCLR and MOCO.  Also, sparsity is not the only metric where this happens -- dimensionality is another, as is invariance to data augmentations.  My main interest is actually in how these models compare to neural responses in visual cortex, and there are systematic differences there as well.

I had considered the input normalization difference --  at one point I tried using the torchvision training code without any input normalization but it didn't seem to make a difference.  So I figure it is some other small detail, perhaps the learning rate schedule.  If any other differences come to mind let me know.  For instance, was a weight decay of 10^-6 used for the supervised networks in this repo?  I think the Torchvision standard is 10^-4.

**Comment by chentingpc** [comment]
the supervised learning models released here are trained using SGD with lr base of 0.1 (linear lr scaling) and weight decay (1e-4), the same as torchvision. the lars and weight decay (1e-6) are only used for simclr training.

**Comment by jlindsey15** [comment]
Got it, thanks -- to confirm, you still use the cosine learning schedule for the supervised models released here?

**Comment by chentingpc** [comment]
yes
